workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/WorkerGenerationJob.cs

```diff
@@ -20,58 +20,85 @@ public class WorkerGenerationJob : CodegenJob
+        public WorkerGenerationJob(string baseOutputDir, IFileSystem fileSystem, DetailsStore detailsStore)
+            : base(baseOutputDir, fileSystem, detailsStore)
         {
+            var jobName = nameof(WorkerGenerationJob);
+            logger.Info($"Initialising {jobName}");
+
             workerTypesToGenerate = ExtractWorkerTypes(CodeGeneratorOptions.Instance.WorkerJsonDirectory);
 
-            OutputFiles.Add(Path.Combine(relativeEditorPath, workerFileName));
-            OutputFiles.Add(Path.Combine(relativeEditorPath, workerListFileName));
-            OutputFiles.Add(Path.Combine(relativeOutputPath, buildSystemFileName));
+            var outputFilePaths = new List<string>
+            {
+                Path.Combine(relativeEditorPath, workerFileName),
+                Path.Combine(relativeEditorPath, workerListFileName),
+                Path.Combine(relativeOutputPath, buildSystemFileName)
+            };
+
+            foreach (var filePath in outputFilePaths)
+            {
+                AddOutputFile(filePath);
+            }
+            logger.Info($"Added {outputFilePaths.Count} job output files");
+
+            logger.Info($"Finished initialising {jobName}");
         }
 
         protected override void RunImpl()
         {
+            logger.Info($"Generating {workerFileName}");
             var unityWorkerMenuGenerator = new UnityWorkerMenuGenerator();
             var workerCode = unityWorkerMenuGenerator.Generate(workerTypesToGenerate);
-            Content.Add(Path.Combine(relativeEditorPath, workerFileName), workerCode);
+            AddContent(Path.Combine(relativeEditorPath, workerFileName), workerCode);
 
+            logger.Info($"Generating {buildSystemFileName}");
             var buildSystemAssemblyGenerator = new BuildSystemAssemblyGenerator();
             var assemblyCode = buildSystemAssemblyGenerator.Generate();
-            Content.Add(Path.Combine(relativeOutputPath, buildSystemFileName), assemblyCode);
+            AddContent(Path.Combine(relativeOutputPath, buildSystemFileName), assemblyCode);
 
-            Content.Add(Path.Combine(relativeEditorPath, workerListFileName), string.Join(Environment.NewLine, workerTypesToGenerate));
+            logger.Info($"Generating {workerListFileName}");
+            AddContent(Path.Combine(relativeEditorPath, workerListFileName), string.Join(Environment.NewLine, workerTypesToGenerate));
         }
 
         private List<string> ExtractWorkerTypes(string path)
         {
+            logger.Info($"Extracting worker types from {path}");
+
             var workerTypes = new List<string>();
-            var fileNames = Directory.EnumerateFiles(path, "*.json");
+
+            var fileNames = Directory.EnumerateFiles(path, "*.json").ToList();
+            logger.Trace($"Found {fileNames.Count()} worker json files: {string.Join(", ", fileNames)}");
```

Comment:
```suggestion
            logger.Trace($"Found {fileNames.Count()} worker json files: {string.Join("\n", fileNames)}");
```

Nit. Maybe more legible?